SQL/0000-00-00-schema.sql

```diff
@@ -688,20 +688,34 @@ CREATE TABLE `mri_protocol` (
   `Center_name` varchar(4) NOT NULL default '',
   `ScannerID` int(10) unsigned NOT NULL default '0',
   `Scan_type` int(10) unsigned NOT NULL default '0',
-  `TR_range` varchar(255) default NULL,
-  `TE_range` varchar(255) default NULL,
-  `TI_range` varchar(255) default NULL,
-  `slice_thickness_range` varchar(255) default NULL,
-  `FoV_x_range` varchar(255) default NULL,
-  `FoV_y_range` varchar(255) default NULL,
-  `FoV_z_range` varchar(255) default NULL,
-  `xspace_range` varchar(255) default NULL,
-  `yspace_range` varchar(255) default NULL,
-  `zspace_range` varchar(255) default NULL,
-  `xstep_range` varchar(255) default NULL,
-  `ystep_range` varchar(255) default NULL,
-  `zstep_range` varchar(255) default NULL,
-  `time_range` varchar(255) default NULL,
+  `TR_min` varchar(10) DEFAULT NULL,
+	`TR_max` varchar(10) DEFAULT NULL,
+	`TE_min` varchar(10) DEFAULT NULL,
+	`TE_max` varchar(10) DEFAULT NULL,
+	`TI_min` varchar(10) DEFAULT NULL,
+	`TI_max` varchar(10) DEFAULT NULL,
+	`slice_thickness_min` varchar(10) DEFAULT NULL,
+	`slice_thickness_max` varchar(10) DEFAULT NULL,
+	`FOV_x_min` varchar(5) DEFAULT NULL,
+	`FOV_x_max` varchar(5) DEFAULT NULL,
+	`FOV_y_min` varchar(5) DEFAULT NULL,
+	`FOV_y_max` varchar(5) DEFAULT NULL,
+	`FOV_z_min` varchar(5) DEFAULT NULL,
+	`FOV_z_max` varchar(5) DEFAULT NULL,
+	`xspace_min` varchar(5) DEFAULT NULL,
+	`xspace_max` varchar(5) DEFAULT NULL,
+	`yspace_min` varchar(5) DEFAULT NULL,
+	`yspace_max` varchar(5) DEFAULT NULL,
+	`zspace_min` varchar(5) DEFAULT NULL,
+	`zspace_max` varchar(5) DEFAULT NULL,
+	`xstep_min` varchar(5) DEFAULT NULL,
+	`xstep_max` varchar(5) DEFAULT NULL,
+        `ystep_min` varchar(5) DEFAULT NULL,
+        `ystep_max` varchar(5) DEFAULT NULL,
+        `zstep_min` varchar(5) DEFAULT NULL,
+        `zstep_max` varchar(5) DEFAULT NULL,
+	`time_min` varchar(50) DEFAULT NULL,
+	`time_max` varchar(50) DEFAULT NULL;
   `series_description_regex` varchar(255) default NULL,
```

Comment:
Should not these value be numerical type?

Reply:
I think they should be a `DECIMAL` type. (Not sure about the precision, though. Talk to @MounaSafiHarab and @LeighMac about what the largest step size might be and how many decimal points are needed.)

Reply:
Indeed, now that they are separate fields these should be decimal:
- for TR, TE, TI => decimal with 4 decimal points (non-negative)
- for *space => allow values up to 1000 to cover a wide range of high resolution sequences maybe? (non-negative). From what I understand, *space corresponds to the number of voxels along a certain axis in the acquisition so it could probably be an integer. @MounaSafiHarab what do you think?
- for *step and slice_thickness => decimal with 4 decimal points (non-negative).
- for time => integer up to 4 digits (non-negative) since it refers to a number of volumes

@MounaSafiHarab can confirm

Reply:
Are those conservative values for the largest value that might exist, or are they what realistically happens? (We should aim to have the magnitude/precision cover a little more than we need to do we don't find ourselves having to change the schema when higher resolution scans/protocols/whatevers come out in the future.)

Reply:
@driusan I tried to be generous in what I suggested those numbers for the same reason you mentioned.

Realistically, 3 digits should be enough but we never know so I added a 4th decimal. For the time, most of them would probably be 3 digits as well but in case someone hates really his participant and let him rot in the scanner, 9999 volumes should be enough ;).

Reply:
> I think they should be a DECIMAL type. (Not sure about the precision, though. Talk to @MounaSafiHarab and @LeighMac about what the largest step size might be and how many decimal points are needed.)

A precision of 4 decimal places is good enough. @nicolasbrossard on the MRI side in 19.0 had a pull request where he sets this precision to the 4th decimal place and assume there is a match for any difference lower than that (https://github.com/aces/Loris-MRI/pull/206).